### PR TITLE
Allow for custom alternate file function

### PR DESCRIPTION
**Pull request by gmartsenkov**
Hopefully this could be helpful for somebody else.
I'm using https://github.com/rgroli/other.nvim and wanted to integrate with it.
Not sure if this needs a test, my first time touching vimscript really :shushing_face: 

Make sure these boxes are checked before submitting your pull request:

- [x] Add fixtures and spec when implementing or updating a test runner
- [x] Update the README accordingly
- [x] Update the Vim documentation in `doc/test.txt`

### CODE REVIEW

**Review by codeinabox — APPROVED** [review]
Great addition! Thank you for taking the time to contribute

**Comment by mbudde** [comment]
I get the error "Funcref variable must start with a capital" when trying to assign a funcref to `test#custom_alternate_file`. Is this a Neovim thing? I'm using Neovim 0.7.2.

File: README.md [hunk 1]
```diff
@@ -610,6 +610,19 @@ You can disable this integration by doing
...
+
+If you are using a different library for jumping between implementation and test file
+you can define a custom function that returns the test filename.
+
+```vim
+function! CustomAlternateFile(cmd)
+  return "test_file_spec.rb"
+endfunction
+
+let g:test#custom_alternate_file = function('echo')
```
Comment (mbudde) [inline README.md:623]: Shouldn't that be `CustomAlternateFile` instead of `echo`?

Comment (codeinabox) [reply README.md:623]: Good question, @gmartsenkov what are your thoughts? Perhaps we need some automated tests around this

**Pull request merged**